tests/phpunit/tests/compat/array_key_last.php

```diff
@@ -0,0 +1,73 @@
+<?php
+
+/**
+ * @group compat
+ *
+ * @covers ::array_key_last
+ */
+class Tests_Compat_array_key_last extends WP_UnitTestCase {
+
+	/**
+	 * Test that array_key_last() is always available (either from PHP or WP).
+	 * @ticket 45055
+	 */
+	public function test_array_key_last_availability() {
+		$this->assertTrue( function_exists( 'array_key_last' ) );
+	}
+
+
+	/**
+	 * @dataProvider array_key_last_provider
+	 * @ticket 45055
+	 */
+	public function test_array_key_last( $expected, $array ) {
```

Comment:
- Suggest renaming this data provider to `data_array_key_last` per [Trac 53119](https://core.trac.wordpress.org/ticket/53119).
- Suggest changing `$array` changed to `$arr` and updating 'array' to 'arr' in the data provider. Avoid using PHP keywords in a method/function signature due to PHP 8.1 and named key parameters in PHP 8.0. As a general best practice, try to avoid them in tests too.
- For tests with data providers, all parameters should be documented.